### PR TITLE
Fix: validate check and probe labels

### LINKED ISSUE #99: agent accepts invalid label names
Prometheus is very strict about the labels it accepts, basically they should match `/^[a-zA-Z][a-zA-Z0-9]*$/`.

We are not strict enough with the labels we are accepting, and that causes Prometheus to reject them. This should be fixed both in the backend code and the front end plugin.

### PR DESCRIPTION
**Pull request by mem**
We are not validating the labels that we are going to apply to the
metrics, and we eventually hit a call to MustRegister that causes the
probes to crash with bad metrics.

Add validation code for labels, that gets called from checks and probes.
This is going to cause the check to be rejected if it makes it to the
probe, but the intention is that it's rejected in the API or the
frontend, where it can be handled more gracefully. The API is going to
inherit this validation.

Also add TestCheckValidate to make sure we are calling these validations
for incoming checks.

Partially fixes #99 (because of required API and frontend fixes).

Signed-off-by: Marcelo E. Magallon <marcelo.magallon@grafana.com>

### CODE REVIEW

File: pkg/pb/synthetic_monitoring/checks_extra.go [hunk 1]
```diff
@@ -97,20 +100,16 @@ func (c *Check) Validate() error {
 
 	// timeout must be in [1, 10] seconds
 	if c.Timeout < 1*1000 || c.Timeout > 10*1000 {
-		return ErrInvalidCheckFrequency
+		return ErrInvalidCheckTimeout
```
Comment (electron0zero) [inline pkg/pb/synthetic_monitoring/checks_extra.go:103]: we should also validate that check frequency is grater then check timeout.

If we allow check frequency to be less then timeout, we would end up running concurrent checks for a target.

issue: https://github.com/grafana/synthetic-monitoring-api/issues/95

File: pkg/pb/synthetic_monitoring/checks_extra.go [hunk 2]
```diff
@@ -236,6 +231,27 @@ func (p *Probe) Validate() error {
...
+	if len(l.Name) == 0 {
+		return ErrInvalidLabelName
+	}
+
+	// This bit is lifted from Prometheus code, except that
+	// Prometheus accepts /^[a-zA-Z_][a-zA-Z0-9_]*$/ and we accept
+	// /^[a-zA-Z0-9_]+$/ because these names are going to be
+	// prefixed with "label_".
+	for _, b := range l.Name {
+		if !((b >= 'a' && b <= 'z') || (b >= 'A' && b <= 'Z') || b == '_' || (b >= '0' && b <= '9')) {
```
Comment (electron0zero) [inline pkg/pb/synthetic_monitoring/checks_extra.go:244]: curious to know why we didn't use https://golang.org/pkg/regexp/ here.

File: pkg/pb/synthetic_monitoring/checks_extra_test.go [hunk 3]
```diff
@@ -281,6 +352,57 @@ func TestValidateHostPort(t *testing.T) {
 	}
 }
 
+func TestValidateLabel(t *testing.T) {
```
Comment (electron0zero) [inline pkg/pb/synthetic_monitoring/checks_extra_test.go:355]: :rocket:

File: pkg/pb/synthetic_monitoring/checks_extra.go (hunk 1 above)
Comment (mem) [reply pkg/pb/synthetic_monitoring/checks_extra.go:103]: thanks, I'll include that here

File: pkg/pb/synthetic_monitoring/checks_extra.go (hunk 2 above)
Comment (mem) [reply pkg/pb/synthetic_monitoring/checks_extra.go:244]: According to Prometheus code it's for performance reasons. I doubt it makes a difference here, they probably run this bit of code many times, we only do it when checks are added.

**Pull request merged**